[RHIDP-5032] Add procedure for deploying the Operator on AKS using OLM (#729)
redhat-developer/red-hat-developers-documentation-rhdh: modules/installation/proc-rhdh-deploy-aks-operator.adoc

[id='proc-rhdh-deploy-aks-operator_{context}']
= Deploying {product-short} on {aks-short} with the Operator

You can install the {product-short} Operator on {aks-short} using the https://olm.operatorframework.io[Operator Lifecycle Manager (OLM) framework]. Following that, you can proceed to deploy your {product-short} instance in {aks-short}.

== Installing the {product-short} Operator

.Prerequisites

* You have cluster-scoped privileges.
* You have set the context to the {aks-short} cluster in your current `kubeconfig`. For more information, see https://learn.microsoft.com/en-us/azure/aks/learn/quick-kubernetes-deploy-cli#connect-to-the-cluster[Connect to the cluster].
* You have installed `kubectl`. For more information, see https://learn.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-install-cli[`az aks install-cli`].
* You have subscribed to `registry.redhat.io`. For more information, see https://access.redhat.com/RegistryAuthentication[{company-name} Container Registry Authentication].

> * You installed the Operator Lifecycle Manager (OLM) framework. See link:https://olm.operatorframework.io/docs/[Operator Lifecycle Manager(OLM)].

Reviewer: `proc-install-operator` is the file that would need to be inserted with `includes` metadata.  

https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/main/modules/installation/proc-install-operator.adoc

> * You installed the Operator Lifecycle Manager (OLM) framework. See link:https://olm.operatorframework.io/docs/[Operator Lifecycle Manager(OLM)].

Author: > We need to take a look at the [Installing the Red Hat Developer Hub Operator](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.4/html/installing_red_hat_developer_hub_on_openshift_container_platform/assembly-install-rhdh-ocp-operator#proc-install-operator_assembly-install-rhdh-ocp-operator) procedure to see if it can be reused in this procedure as-is. We want to single-source whenever possible.
> [...]
> We already have a separate procedure module for this that is applicable to OCP, EKS, and AKS but is currently only used in the OCP doc. We need to insert it here with metadata rather than rewrite it here this way.

@linfraze [Installing the Red Hat Developer Hub Operator](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.4/html/installing_red_hat_developer_hub_on_openshift_container_platform/assembly-install-rhdh-ocp-operator#proc-install-operator_assembly-install-rhdh-ocp-operator) is really specific to OCP in my opinion. It leverages the OCP web console, which is not available on non-OCP platforms. Also, the prerequisites are different on OCP vs AKS/GKE/EKS. For example, OLM is already available OOTB on OCP, but needs to be installed on the other platforms. Same point for pulling images from `registry.redhat.io`.
So I am not sure we can reuse that content from there.
I think it could have been possible if we had a procedure for installing the Operator on OCP from the CLI.
Did I miss anything?

> * You installed the Operator Lifecycle Manager (OLM) framework. See link:https://olm.operatorframework.io/docs/[Operator Lifecycle Manager(OLM)].

Reviewer: Hm, good to note about the CLI Operator install... 
I did notice a lot more Prerequs for AKS/EKS than for OCP. 
While we do need to put this new AKS/EKS-specific procedure in a separate, reusable file so that we adhere to  modularity and single-sourcing best practices, I want to hear others' thoughts (@Preeticp) on whether we let this PR go out as-is and then open a new PR that creates and `includes` a separate proc file for installing the Operator for AKS/EKS or if we should go ahead and correct it now.

> * You installed the Operator Lifecycle Manager (OLM) framework. See link:https://olm.operatorframework.io/docs/[Operator Lifecycle Manager(OLM)].

Author: /cc @Preeticp 
/assign @Preeticp

[id='proc-rhdh-deploy-aks-operator_{context}']
= Deploying {product-short} on {aks-short} with the Operator

> The {product} Operator installation requires the Operator Lifecycle Manager (OLM) framework.

Reviewer 2: Consider moving to the Prerequisites section.

```suggestion
```

Author: Fixed: https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/729#discussion_r1867617883

> .Additonal resources
> * For information about the OLM, see link:https://olm.operatorframework.io/docs/[Operator Lifecycle Manager(OLM)] documentation.

Reviewer 2: To respect modular docs templates, and for minimalism:

```suggestion
.Prerequisites
* You installed the Operator Lifecycle Manager (OLM) framework. See link:https://olm.operatorframework.io/docs/[Operator Lifecycle Manager(OLM)].
```

Reviewer 2: Or it seems, leftovers?

```suggestion
```

> == Installing the {product-short} Operator with the OLM framework

Reviewer 2: Leftover?

```suggestion
```

Author: Fixed: https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/729#discussion_r1867617883

You can install the {product-short} Operator on {aks-short} using the https://olm.operatorframework.io[Operator Lifecycle Manager (OLM) framework]. Following that, you can proceed to deploy your {product-short} instance in {aks-short}.

.Prerequisites

* You have cluster-scoped privileges.
* You have set the context to the {aks-short} cluster in your current `kubeconfig`. For more information, see https://learn.microsoft.com/en-us/azure/aks/learn/quick-kubernetes-deploy-cli#connect-to-the-cluster[Connect to the cluster].

> * You have installed `kubectl`. For more information, see https://learn.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-install-cli[`az aks install-cli].

Reviewer 2: same typo 
```suggestion
* You have installed `kubectl`. For more information, see https://learn.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-install-cli[`az aks install-cli`].
```
